src/Microsoft.AspNetCore.Routing/Matchers/ZeroEntryJumpTable.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    internal class ZeroEntryJumpTable : JumpTable
+    {
+        private readonly int _defaultDestination;
+        private readonly int _exitDestination;
+
+        public ZeroEntryJumpTable(int defaultDestination, int exitDestination)
+        {
+            _defaultDestination = defaultDestination;
+            _exitDestination = exitDestination;
+        }
+
+        public unsafe override int GetDestination(string path, PathSegment segment)
+        {
+            return segment.Length == 0 ? _exitDestination : _defaultDestination;
```

Comment:
Maybe add a comment here about why segment length is checked

Reply:
This is explained on `JumpTableBuilder.ExitDestination`. I don't really want the same comment to appear in 5 other places.